tests/snippets/Firestore/WriteBatchTest.php

```diff
@@ -0,0 +1,180 @@
+
+namespace Google\Cloud\Tests\Snippets\Firestore;
+
+use Prophecy\Argument;
+use Google\Cloud\Firestore\WriteBatch;
+use Google\Cloud\Firestore\ValueMapper;
+use Google\Cloud\Dev\Snippet\Parser\Snippet;
+use Google\Cloud\Dev\Snippet\SnippetTestCase;
+use Google\Cloud\Firestore\Connection\ConnectionInterface;
+
+/**
+ * @group firestore
+ * @group firestore-writebatch
+ */
+class WriteBatchTest extends SnippetTestCase
+{
+    const DATABASE = 'projects/example_project/databases/(default)';
+    const DOCUMENT = 'projects/example_project/databases/(default)/documents/a/b';
+
+    private $connection;
+    private $batch;
+
+    public function setUp()
+    {
+        $this->connection = $this->prophesize(ConnectionInterface::class);
+        $this->batch = \Google\Cloud\Dev\stub(WriteBatch::class, [
+            $this->connection->reveal(),
+            new ValueMapper($this->connection->reveal(), false),
+            self::DATABASE
+        ], ['connection', 'transaction']);
+    }
+
+    public function testClass()
+    {
+        $snippet = $this->snippetFromClass(WriteBatch::class);
+        $res = $snippet->invoke('batch');
+        $this->assertInstanceOf(WriteBatch::class, $res->returnVal());
+    }
+
+    public function testCreate()
+    {
+        $snippet = $this->snippetFromMethod(WriteBatch::class, 'create');
+        $this->commitAndAssert($snippet, [
+            [
+                'updateMask' => ['name'],
+                'currentDocument' => ['exists' => false],
+                'update' => [
+                    'name' => self::DOCUMENT,
+                    'fields' => [
+                        'name' => ['stringValue' => 'John']
+                    ]
+                ]
+            ]
+        ]);
+    }
+
+    public function testUpdate()
+    {
+        $snippet = $this->snippetFromMethod(WriteBatch::class, 'update');
+        $this->commitAndAssert($snippet, [
+            [
+                'updateMask' => ['name'],
+                'currentDocument' => ['exists' => true],
+                'update' => [
+                    'name' => self::DOCUMENT,
+                    'fields' => [
+                        'name' => ['stringValue' => 'John']
+                    ]
+                ]
+            ]
+        ]);
+    }
+
+    public function testSet()
+    {
+        $snippet = $this->snippetFromMethod(WriteBatch::class, 'set');
+        $this->commitAndAssert($snippet, [
+            [
+                'update' => [
+                    'name' => self::DOCUMENT,
+                    'fields' => [
+                        'name' => ['stringValue' => 'John']
+                    ]
+                ]
+            ]
+        ]);
+    }
+
+    public function testDelete()
+    {
+        $snippet = $this->snippetFromMethod(WriteBatch::class, 'delete');
+        $this->commitAndAssert($snippet, [
+            [
+                'delete' => self::DOCUMENT
+            ]
+        ]);
+    }
+
+    public function testVerify()
```

Comment:
Verify is no longer part of the public API. This should go away when you regenerate the client.